<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ArtistEdit</class>
 <widget class="QDialog" name="ArtistEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>662</width>
    <height>794</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <bold>true</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>Artist Editor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../resources.qrc">
    <normaloff>:/images/icon.ico</normaloff>:/images/icon.ico</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color:white</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="tabShape">
      <enum>QTabWidget::TabShape::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <property name="elideMode">
      <enum>Qt::TextElideMode::ElideNone</enum>
     </property>
     <property name="usesScrollButtons">
      <bool>true</bool>
     </property>
     <property name="documentMode">
      <bool>false</bool>
     </property>
     <property name="tabBarAutoHide">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="infoWidget">
      <attribute name="title">
       <string>Profile</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="2" column="0" colspan="2">
        <widget class="ImageView" name="photoView" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <bold>false</bold>
          </font>
         </property>
         <property name="acceptDrops">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QPushButton" name="removeCoverButton">
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/dialogs/delete.png</normaloff>:/images/dialogs/delete.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="6" column="0" colspan="2">
        <widget class="QTextEdit" name="biographyEdit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::WinPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Plain</enum>
         </property>
         <property name="placeholderText">
          <string>Type the biography here...</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="nameEdit">
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="placeholderText">
          <string>Type the artist's name here</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label">
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Name:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="2">
        <widget class="QLabel" name="label_2">
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Biography:</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="playlistWidget">
      <attribute name="title">
       <string>Albums</string>
      </attribute>
      <layout class="QVBoxLayout" name="playlistLayout">
       <item>
        <widget class="QLabel" name="notFoundLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <pointsize>15</pointsize>
           <fontweight>ExtraLight</fontweight>
          </font>
         </property>
         <property name="text">
          <string>Oops! No albums found for this artist!</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListView" name="playlistView">
         <property name="frameShape">
          <enum>QFrame::Shape::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Plain</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SelectionMode::ExtendedSelection</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="controlLayout">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item>
          <spacer name="leftSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="addButton">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../resources.qrc">
             <normaloff>:/images/dialogs/add.png</normaloff>:/images/dialogs/add.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>32</width>
             <height>32</height>
            </size>
           </property>
           <property name="flat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="removeButton">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../resources.qrc">
             <normaloff>:/images/dialogs/delete.png</normaloff>:/images/dialogs/delete.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>32</width>
             <height>32</height>
            </size>
           </property>
           <property name="flat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="rightSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImageView</class>
   <extends>QWidget</extends>
   <header>include/components/imageview.h</header>
   <container>1</container>
   <slots>
    <signal>viewActivated()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ArtistEdit</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>494</x>
     <y>738</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>849</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ArtistEdit</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>604</x>
     <y>738</y>
    </hint>
    <hint type="destinationlabel">
     <x>627</x>
     <y>847</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>ArtistEdit</receiver>
   <slot>addPlaylist()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>686</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>859</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeButton</sender>
   <signal>clicked()</signal>
   <receiver>ArtistEdit</receiver>
   <slot>removePlaylist()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>372</x>
     <y>686</y>
    </hint>
    <hint type="destinationlabel">
     <x>7</x>
     <y>846</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playlistView</sender>
   <signal>activated(QModelIndex)</signal>
   <receiver>ArtistEdit</receiver>
   <slot>editPlaylist(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>359</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>655</x>
     <y>98</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeCoverButton</sender>
   <signal>clicked()</signal>
   <receiver>ArtistEdit</receiver>
   <slot>removeCover()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>332</x>
     <y>470</y>
    </hint>
    <hint type="destinationlabel">
     <x>657</x>
     <y>457</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>editPlaylist(QModelIndex)</slot>
  <slot>addPlaylist()</slot>
  <slot>removePlaylist()</slot>
  <slot>removeCover()</slot>
 </slots>
</ui>
